### PR TITLE
cql3: fix a few misformatted printouts of column names in error messages

### LINKED ISSUE #13657: Misformatted printout of column name in LWT error message
When attempting the LWT operation `set m['foo'] = 'bar'` when m is a **frozen** collection, Scylla wants to print the message:

```
Invalid operation (m['foo'] = 'bar') for frozen collection column m
```

But it ends up printing the hexadecimal ASCII (!) of the column name:
```
Invalid operation (m['foo'] = 'bar') for frozen collection column 6d
```

I guess we end up printing a "bytes" instead of a string.

Reproduced by translated Cassandra unit test `cassandra_tests/validation/operations/insert_update_if_condition_collections_test.py`::`testMapItem`

### PR DESCRIPTION
**Pull request by nyh**
Fix a few cases where instead of printing column names in error messages, we printed weird stuff like ASCII codes or the address of the name.

Fixes #13657

### CODE REVIEW

File: cql3/operation.cc [hunk 1]
```diff
@@ -343,9 +343,9 @@ operation::element_deletion::affected_column() const {
 shared_ptr<operation>
 operation::element_deletion::prepare(data_dictionary::database db, const sstring& keyspace, const column_definition& receiver) const {
     if (!receiver.type->is_collection()) {
-        throw exceptions::invalid_request_exception(format("Invalid deletion operation for non collection column {}", receiver.name()));
+        throw exceptions::invalid_request_exception(format("Invalid deletion operation for non collection column {}", receiver.name_as_text()));
     } else if (!receiver.type->is_multi_cell()) {
-        throw exceptions::invalid_request_exception(format("Invalid deletion operation for frozen collection column {}", receiver.name()));
+        throw exceptions::invalid_request_exception(format("Invalid deletion operation for frozen collection column {}", receiver.name_as_text()));
```
Comment (avikivity) [inline cql3/operation.cc:348]: We should have a formatter for receiver that makes sense everywhere. But won't delay this fix.

**Review by avikivity — APPROVED** [review]

File: cql3/operation.cc (hunk 1 above)
Comment (nyh) [reply cql3/operation.cc:348]: There are several ways to print "receiver" (the full operation? just the name) and receiver.name() indicates we want just the name. Or so thought the person who wrote this code which forgot (in a few places) the distinction between receiver.name( ) name_as_text().

What bothers me more is that we have silly formatters for "bytes" and "shared_ptr" - and I think that almost in no cases do we really want to print ASCII codes or a pointer - and almost every place we did this, it was unintentional. Personally I would have liked these formatters either to not exist or do something reasonable, and in the rare cases where somebody really wants to print a pointer or ASCII, we could have an explicit convertor for that (e.g., to_hex(bytes), or to_address(ptr)).

**Comment by nyh** [comment]
Amusing, CI failed because the test/cql "approval tests" expected lines like
```
Error from server: code=2200 [Invalid query] message="Invalid operation (s = {true} + s) for non list column 73"
```
as the gold truth, but this patch changes it (fixes it!) to
```
Error from server: code=2200 [Invalid query] message="Invalid operation (s = {true} + s) for non list column s"
```

I'll fix the expected test values to be correct to make the CI pass (it's easy).

File: cql3/operation.cc (hunk 1 above)
Comment (avikivity) [reply cql3/operation.cc:348]: formatters can have modifiers that say how you want it printed.

Comment (nyh) [reply cql3/operation.cc:348]: I'm not a fan of that. C++ has methods, it's complicated enough I think it's good enough, we don't need to start adding "mixins" in the form of formatter modifiers for specific classes.

**Comment by nyh** [comment]
Pushed new version fixing the wrong error messages in the saved output files in the approval tests. Let's hope it passes CI now.

**Review by denesb — APPROVED** [review]

**Comment by nyh** [comment]
@denesb pushed new version fixing additional test which I missed in the last round for "approval" tests (test/cql) which enshrined the wrong behavior.
Let's wait for the CI to confirm I fixed it all.

**Comment by denesb** [comment]
> @denesb I think you merged this version with CI still failing :-(

Oops, sorry about that.

>  Let's wait for the CI to confirm I fixed it all.

:+1:

**Comment by nyh** [comment]
Finally I got all the tests to pass. Sorry about the earlier mess Botond.

**Comment by denesb** [comment]
> Finally I got all the tests to pass. Sorry about the earlier mess Botond.

No problem, we all mess up sometimes.

Queued.

**Pull request merged**